Extension/src/LanguageServer/settings.ts

```diff
@@ -178,43 +180,51 @@ class Settings {
 
     protected get Section(): vscode.WorkspaceConfiguration { return this.settings; }
 
-    protected getWithFallback<T>(section: string, deprecatedSection: string): T {
-        const info: any = this.settings.inspect<T>(section);
+    // If the setting has an undefined default, look for the workspaceFolder, workspace and global values as well.
+    public getArrayOfStringsWithUndefinedDefault(section: string): string[] | undefined;
+    public getArrayOfStringsWithUndefinedDefault(section: string, allowNull: boolean): string[] | undefined | null;
+    public getArrayOfStringsWithUndefinedDefault(section: string, allowNull: boolean = false): string[] | undefined | null {
+        const info: any = this.settings.inspect<string[]>(section);
+        if (allowNull && info === null) {
+            return null;
+        }
         if (info.workspaceFolderValue !== undefined) {
-            return info.workspaceFolderValue;
+            if (isArrayOfString(info.workspaceFolderValue)) {
+                return info.workspaceFolderValue;
+            }
         } else if (info.workspaceValue !== undefined) {
-            return info.workspaceValue;
+            if (isArrayOfString(info.workspaceValue)) {
+                return info.workspaceValue;
+            }
         } else if (info.globalValue !== undefined) {
-            return info.globalValue;
-        }
-        const value: T | undefined = this.settings.get<T>(deprecatedSection);
-        if (value !== undefined) {
-            return value;
+            if (isArrayOfString(info.globalValue)) {
+                return info.globalValue;
+            }
         }
-        return info.defaultValue;
+        return undefined;
     }
 
-    protected getWithNullAsUndefined<T>(section: string): T | undefined {
-        const result: T | undefined | null = this.settings.get<T>(section);
-        if (result === null) {
-            return undefined;
-        }
-        return result;
-    }
+    public getStringWithUndefinedDefault(section: string): string | undefined {
+        const info: any = this.settings.inspect<string>(section);
 
-    public getWithUndefinedDefault<T>(section: string): T | undefined {
-        const info: any = this.settings.inspect<T>(section);
         if (info.workspaceFolderValue !== undefined) {
-            return info.workspaceFolderValue;
+            if (isString(info.workspaceFolderValue)) {
```

Comment:
I don't think checks for undefined are correct here. If you enter this branch and the `isString` fails, it won't try looking at `info.workspaceValue` or `info.globalValue`

Reply:
That's a good point, perhaps there was a bug in the existing behavior. I think making these all single if statements would be the correct approach.